Mixpanel/MixpanelNotificationServiceExtension.swift

```diff
@@ -0,0 +1,125 @@
+import UserNotifications
+
+private let dynamicCategoryIdentifier = "MP_DYNAMIC"
+private let mediaUrlKey = "mp_media_url"
+
+@available(iOS 11.0, *)
+open class MixpanelNotificationServiceExtension: UNNotificationServiceExtension {
+    open override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
+        guard let bestAttemptContent = (request.content.mutableCopy() as? UNMutableNotificationContent) else {
+            contentHandler(request.content)
+            return
+        }
+        
+        self.maybeAttachButtons(bestAttemptContent: bestAttemptContent) {
+            self.maybeAttachMedia(bestAttemptContent: bestAttemptContent) {
+                contentHandler(bestAttemptContent)
+            }
+        }
+    }
+    
+    func maybeAttachButtons(bestAttemptContent: UNMutableNotificationContent, completionHandler: @escaping () -> Void) {
+        guard let buttons = bestAttemptContent.userInfo["mp_buttons"] as? [[AnyHashable: Any]] else {
+            NSLog("maybeAttachButtons: No action buttons found in the push notification payload.")
```

Comment:
I realize that `NSLog` might not be the correct way to log this debugging info but trying to use `print` or the `MPLog` stuff resulted in nothing showing up. `NSLog` was the only way I was able to see output in the console.